hooks/open-telemetry/src/main/java/dev/openfeature/contrib/hooks/otel/OpenTelemetryMetricHook.java

```diff
@@ -0,0 +1,167 @@
+package dev.openfeature.contrib.hooks.otel;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.FlagEvaluationDetails;
+import dev.openfeature.sdk.FlagMetadata;
+import dev.openfeature.sdk.Hook;
+import dev.openfeature.sdk.HookContext;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.LongUpDownCounter;
+import io.opentelemetry.api.metrics.Meter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static dev.openfeature.contrib.hooks.otel.OtelCommons.ERROR_KEY;
+import static dev.openfeature.contrib.hooks.otel.OtelCommons.REASON_KEY;
+import static dev.openfeature.contrib.hooks.otel.OtelCommons.flagKeyAttributeKey;
+import static dev.openfeature.contrib.hooks.otel.OtelCommons.providerNameAttributeKey;
+import static dev.openfeature.contrib.hooks.otel.OtelCommons.variantAttributeKey;
+
+
+/**
+ * OpenTelemetry metric hook records metrics at different {@link Hook} stages.
+ */
+@Slf4j
+@SuppressWarnings("PMD.TooManyStaticImports")
+public class OpenTelemetryMetricHook implements Hook {
```

Comment:
```suggestion
public class MetricsHook implements Hook {
```

I think `dev.openfeature.contrib.hooks.otel` conveys well-enough this is an OTel component, so we don't need it in the class name.

Reply:
Agree with this proposal 😞  but I think we already have the OpenTelemetry prefix in existing classes [1]. Changing only this is odd.

[1] - https://github.com/open-feature/java-sdk-contrib/tree/main/hooks/open-telemetry/src/main/java/dev/openfeature/contrib/hooks/otel

Reply:
> Agree with this proposal disappointed but I think we already have the OpenTelemetry prefix in existing classes [1]. Changing only this is odd.
> 
> [1] - https://github.com/open-feature/java-sdk-contrib/tree/main/hooks/open-telemetry/src/main/java/dev/openfeature/contrib/hooks/otel

Ya I think we should release a 2.0 and rename them all. That would be my vote.

Reply:
Do you suggest doing that along with this change 🤔  Or should we first release this and then follow up on the v2.0

Reply:
I don't think it needs to be done with this change, but I think it should be done before this change is released. I think having an `OpenTelemetryHook` and a `OpenTelemetryMetricHook` is inherently confusing - it's not clear what the first one does.

Reply:
Agree. I would like a follow-up as this PR focus on a feature. And the follow-up is a breaking change with a refactoring

Reply:
I will do this change with this PR and bump the version to 2.0.0

Reply:
Fine with me!

Reply:
Done - https://github.com/open-feature/java-sdk-contrib/blob/4c6f93da9e27cdc87cf666c996a0fdc23ce5cab9/hooks/open-telemetry/src/main/java/dev/openfeature/contrib/hooks/otel/MetricsHook.java